editor.planx.uk/src/@planx/components/MapAndLabel/Public/Context.tsx

```diff
@@ -99,11 +99,14 @@ export const MapAndLabelProvider: React.FC<MapAndLabelProviderProps> = (
   });
 
   const [activeIndex, setActiveIndex] = useState<number>(
-    previousFormData?.length - 1 || -1,
+    previousFormData ? previousFormData?.length - 1 : 0,
   );
 
   const [minError, setMinError] = useState<boolean>(false);
   const [maxError, setMaxError] = useState<boolean>(false);
+  const [loadPreviousValues, setLoadPreviousValues] = useState<boolean>(
```

Comment:
Went round the blocks with this trying to find an elegant solution, this felt like the best option to ensure previous values only loaded once

Reply:
From what I could understand, `previouslySubmittedData` is always present once you navigate backwards, so when you add a new point the code will run again and it will add the new point to `features` then hit the `if()` statement then reassign `features` to be `previouslySubmittedData.data...`

I initially tried to place it in a `useEffect()` to only run the `if` statement on initial mount then whenever `previouslySubmittedData` changed, but I couldn't get it to work so easily, and it felt like a larger change than the state variable.

Have I missed a simpler solution?

Reply:
Thanks for explanation - happy to keep this approach then but can we please rename to `loadPreviousValuesOnMap` for clarity? Because it's never actually handling the form/tabs part based on how I'm reading it !